### ISSUE TITLE
RNFSManager doesn't work in vNext

### ISSUE THREAD

**Issue by namrog84** [microsoft/react-native-windows#2829]
Cannot read property 'RNFSFileTypeRegular' of undefined

react-native-fs has a RNFSManager which has some constants and other fairly low level stuff.
There is a /windows module but it is using cs and is designed for RNWcurrent and so RNWvNext and it's use of 'uwp' makes this not work.

https://github.com/itinance/react-native-fs/tree/master/windows/RNFS

**Comment by harinikmsft** [comment]
Thanks for trying this out. We are currently in the middle of providing the native modules/view managers story for targeting RNW-vnext. So, any of the community modules that need this story will not currently support RNW-vnext yet. Once this story is ready, we would love to get some help making these community modules work against RNW -vnext. 

Stay tuned for updates!

**Comment by stmoy** [comment]
https://github.com/itinance/react-native-fs/pull/868

**Comment by namrog84** [comment]
Unfortunately that looks like its C# only at the moment :( 

But I am super glad any progress is being done on it!

**Comment by gillpeacegood** [comment]
Yeah I made that change. Really aimed for the quickest and easiest port of the old windows version.

Now I think about it much of react-native-fs could probably be implemented with standard c++ library functionality and also the ios and android implementations could potentially use that (maybe via turbo modules?)

**Comment by harshadak2020** [comment]
+1 Another AppConsult customer is asking for this in order to move their project from the legacy C# core to the new C++ one

**Comment by vmoroz** [comment]
Just to provide an update here. @avmoroz is doing a "virtual internship" this summer. The react-native-fs for RNW is almost done. He just needs to complete file upload (download works), and do some clean up such as moving files to the right place. He has also implemented unit tests and a small demo app. See: https://github.com/avmoroz/react-native-fs/tree/dev

**Comment by penx** [comment]
thanks for the update! Any chance this will also be in react native for mac?

**Comment by exotexot** [comment]
@vmoroz Really appreciate the work from @avmoroz for RNFS. I tested his demo app and it works well! Unfortunately not just the upload, but also the download is missing. Do you know when those feature will be available? I'm really needing this! :)

**Comment by avmoroz** [comment]
Hi exotexot,

I took a look at your issue and tried using readFile(), copyFile(), and moveFile() with a 1.21gb ISO file, and wasn't running into those issues, but I want to investigate your issue further.

Based on the information provided, my first guess is that there are some issues with the arguments passed to the to rnfs. If it's possible, I'd like to try and get some more information about what types of files you were passing and how these methods are being called. For example, I know that using readFile with .7z files leads to a strange hang.

If you would like to contact me about this on something other than Github, feel free to send a message to my Appa#5329 on Discord.

**Comment by exotexot** [comment]
Hi @avmoroz,
I've send you a friend-request on discord. Ill try and be online for the next days in order to catch you eventually.

But to push this further:
I can reproduce the error I was describing by using your Demo app found here: https://github.com/avmoroz/react-native-fs/tree/dev by reading the 100MB.bin file found here: https://speed.hetzner.de/ as a base64 encoded string.

**Comment by mudaligekosala** [comment]
Hi @avmoroz I tried to add your project to my existing application,
yarn add https://github.com/avmoroz/react-native-fs.git#dev
pch.h file
#include "winrt/RNFSWinV2.h"
App.cpp
PackageProviders().Append(winrt::RNFSWinV2::ReactPackageProvider());

are these steps correct? I am getting these errors

Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0135	namespace "winrt::RNFSWinV2" has no member "ReactPackageProvider"	UCMobileApp	C:\Projects\mobile-app\UCApp\windows\UCMobileApp\App.cpp	50

**Comment by exotexot** [comment]
Sorry I must have overseen this.

I'm not using RNFS like you described.
I copied RNFSManager.cpp + RNFSManager.h to:
<projectname>/windows/<projectname>/ - folder (where e.g. your App.cpp is located).

Also make sure to add it to your solution.

**Comment by mudaligekosala** [comment]
> Sorry I must have overseen this.
> 
> I'm not using RNFS like you described.
> I copied RNFSManager.cpp + RNFSManager.h to:
> projectname/windows/projectname/ - folder (where e.g. your App.cpp is located).
> 
> Then add them also to your solution file in VS
> 
> And also make sure to remove those entries in pch.h and App.cpp

@exotexot ,Do i need to install rnfs lib using `yarn add https://github.com/avmoroz/react-native-fs.git#dev`  or just copy past the files.
Can correct me if i am wrong here ,
Solution Explore -Right click on solution =>Add=> Existing Item

**Comment by exotexot** [comment]
,Do i need to install rnfs lib using yarn add https://github.com/avmoroz/react-native-fs.git#dev
> Yes. Because the package provides the mapping of react-native functions to the native c++ code.

Solution Explore -Right click on solution =>Add=> Existing Item
> Thats correct!

**Comment by exotexot** [comment]
I may also need to add that im running react-native 0.62.2 and react-native windows 0.62.12

**Comment by mudaligekosala** [comment]
> I may also need to add that im running react-native 0.62.2 and react-native windows 0.62.12
This is my packages,
`"dependencies": {
    "react": "16.11.0",
    "react-native": "^0.62.0-0",
    "react-native-fs": "https://github.com/avmoroz/react-native-fs.git#dev",
    "react-native-windows": "^0.62.0-0"
  }`
then i added those 2 file to my app's` windows/appName `,
this is my Solution Explorer
![image](https://user-images.githubusercontent.com/34033702/96190997-e9f74080-0f8e-11eb-8586-1c0b9c18eefd.png)

but still getting 
![image](https://user-images.githubusercontent.com/34033702/96191024-fbd8e380-0f8e-11eb-9b4e-98590abb9e23.png)

**Comment by exotexot** [comment]
You located the files correctly but made the wrong reference. Sorry, I've corrected my previous post.

Do not add the .cpp +.h to the solution but to the project within your solution with your projectname (Universal Windows) - which references e.g. C:\Projects\mobile-app\UCApp\windows\UCMobileApp\UCMobileApp.vcxproj

Like so: 
![solutionex](https://user-images.githubusercontent.com/10153413/96191490-7defea80-0f44-11eb-811d-ca078f4b6963.PNG)

**Comment by mudaligekosala** [comment]
@exotexot Thank you very much for your help. This is working as expected now.

**Comment by chzappsinc** [comment]
I am getting error
buildd failed with message C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(460,5): error MSB8020: The build tools for Visual Studio 2015 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install Visual Studio 2015 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution".`

But i have installed 2015 Build Tools

![image](https://user-images.githubusercontent.com/65512279/187048497-117b16ec-373f-47ce-bbbf-7dc9ff0fc78e.png)

**Comment by chzappsinc** [comment]
@avmoroz @vmoroz @exotexot 
Any solutions for above error? 

Without re target manually?

Already downloaded 140 2015 tool kit

**Comment by namrog84** [comment]
Which project is failing to build? Your error message is incomplete.
Also, you should start a new issue post about your issue.

https://github.com/microsoft/react-native-windows/issues

**Comment by chzappsinc** [comment]
![Screenshot 2022-08-28 024856](https://user-images.githubusercontent.com/65512279/187090423-d7f5d307-8a01-46e9-9077-e806fee2eb35.png)

**Comment by namrog84** [comment]
If you open up the RNFS.vcxproj

You will find these lines
```
    <PlatformToolset>v140</PlatformToolset>
    <PlatformToolset Condition="'$(VisualStudioVersion)' == '15.0'">v141</PlatformToolset>
    <PlatformToolset Condition="'$(VisualStudioVersion)' == '16.0'">v142</PlatformToolset>
```

You are using VS2022 which is version 17 so it's falling back to v140. 

Add the following line after the above

`<PlatformToolset Condition="'$(VisualStudioVersion)' == '17.0'">v143</PlatformToolset>`

Then it should use the VS2022 build tools (v143) and you dont have to worry about those old 2015 tools anymore.

**Comment by chzappsinc** [comment]
Thanks, @namrog84 It worked!

I need one more help, How to set the title bar color? and force dark mode only!

Sorry if it's not good to ask here, Can you give me any contact details?

**Comment by cjhines** [comment]
I'm getting the following error while calling `downloadFile` with `avmoroz/react-native-fs`. Anyone encountered it or can offer some advice?

```
RNFS.dll!winrt::impl::consume_Windows_Foundation_IPropertyValue<winrt::Windows::Foundation::IReference<unsigned __int64>>::Type() Line 190	C++
RNFS.dll!RNFSManager::ProcessDownloadRequestAsync$_ResumeCoro$1() Line 954	C++
[External Code]	
RNFS.dll!winrt::impl::resume_apartment(const winrt::impl::resume_apartment_context & context, std::experimental::coroutine_handle<void> handle) Line 8289	C++
RNFS.dll!winrt::impl::disconnect_aware_handler::Complete() Line 3148	C++
RNFS.dll!winrt::impl::disconnect_aware_handler::operator()() Line 3139	C++
RNFS.dll!<lambda_7b3a35664967a00a906a66b76e011559>::operator()<winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Windows::Web::Http::HttpResponseMessage,winrt::Windows::Web::Http::HttpProgress> const &,enum winrt::Windows::Foundation::AsyncStatus>(const winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Windows::Web::Http::HttpResponseMessage,winrt::Windows::Web::Http::HttpProgress> & __formal, winrt::Windows::Foundation::AsyncStatus operation_status) Line 3168	C++
RNFS.dll!winrt::impl::delegate<winrt::Windows::Foundation::AsyncOperationWithProgressCompletedHandler<winrt::Windows::Web::Http::HttpResponseMessage,winrt::Windows::Web::Http::HttpProgress>,<lambda_7b3a35664967a00a906a66b76e011559>>::Invoke(void * asyncInfo, int asyncStatus) Line 860	C++
[External Code]
```

<img width="942" alt="Screenshot 2022-09-05 at 17 53 50" src="https://user-images.githubusercontent.com/16542421/188485450-fb461505-6d2f-4e66-8343-93d93880690a.png">

```ts
    const download = await FileSystem.downloadFile({
      fromUrl,
      toFile: destinationPath,
      headers: {
        Authorization: `Bearer ${Config.API_KEY}`,
      },
    });
```

**Comment by exotexot** [comment]
Youre calling the wrong class. It needs to be RNFS.downloadFile(...)

**Comment by cjhines** [comment]
It's just an import alias.

`import FileSystem from 'react-native-fs';`

`RNFS` doesn't work either.

**Comment by birdofpreyru** [comment]
Hi @exotexot 👋 Nice to stumble upon a familiar face here 😃 I wonder, do you know what happened here, and why Windows is not yet supported in `react-native-fs` out of the box? At the first glance on this thread, it seems the working implementation for Windows was ready more than two years ago, and even if it has some deficiencies, it sure works better than the original `react-native-fs`, which just crashes an app straight away if imported. Why was not it merged into `react-native-fs`, is that lib also abandoned by maintainers and thus merits to be forked / hijacked? 🏴‍☠️

**Comment by exotexot** [comment]
@birdofpreyru yes, the original repo is pretty much unmaintained....
